server/src/main/java/org/elasticsearch/transport/TransportService.java

```diff
@@ -974,12 +965,16 @@ public void onAfter() {
             }
 
         } catch (Exception e) {
-            try {
-                channel.sendResponse(e);
-            } catch (Exception inner) {
-                inner.addSuppressed(e);
-                logger.warn(() -> new ParameterizedMessage("failed to notify channel of error message for action [{}]", action), inner);
-            }
+            handleSendToLocalException(channel, e, action);
```

Comment:
Is this ever expected? If not, we should `assert false` here. If we do expect it sometimes I think we should fork to the appropriate executor too.

Reply:
This was handling the exceptions from the `SAME` path which now that you ask is really confusing. I added a dedicated try-catch for that now and an assert here.

Reply:
Oh right so it was. Yes I prefer this now.